sdk/lib/rtl/process.c

```diff
@@ -493,3 +493,19 @@ RtlGetCurrentProcessorNumber(VOID)
     /* Forward to kernel */
     return NtGetCurrentProcessorNumber();
 }
+
+ULONG
+NTAPI
+RtlGetAppcompatVersion(VOID)
+{
+    /* Get the current PEB */
+    PPEB Peb = RtlGetCurrentPeb();
+    if (Peb == NULL)
+    {
+        /* Default to Server 2003 */
+        return _WIN32_WINNT_WS03;
+    }
+
+    /* Calculate OS version from PEB fields */
+    return (Peb->OSMajorVersion << 8) | Peb->OSMinorVersion;
```

Comment:
I think in kernel mode we need to probe and SEH-protect this.

Reply:
Correct.

Reply:
There is no need to "probe" it, since the pointer comes from kernel mode.
```
PPEB
NTAPI
RtlGetCurrentPeb(VOID)
{
    return ((PEPROCESS)(KeGetCurrentThread()->ApcState.Process))->Peb;
}
```
I also don't see why SEH would be needed, since the Peb is a "NoChange" allocation and you can't "free" it or change it's protection (see MiCreatePebOrTeb()).